.github/workflows/codeql-java-analysis.yml

```diff
@@ -70,17 +70,34 @@ jobs:
         # queries: ./path/to/local/query, your-org/your-repo/queries@main
         queries: +security-and-quality
 
+    - name: 'Setup Temurin JDK 8, 11, 17 & 21'
+      uses: actions/setup-java@387ac29b308b003ca37ba93a6cab5eb57c8f5f93 # v4.0.0
+      with:
+        distribution: 'temurin'
+        java-version: |
+          8
+          11
+          17
+          21
```

Comment:
Do we need several JDKs for CodeQL analysis ?
CodeQL is ignored most of the time anyway

Update: OK, I think I understand it - we need them for toolchains.xml

Reply:
Yes, the build checks if all needed JDK versions for the full build are present early on in the process ("fail fast").